pkg/apb/executor.go

```diff
@@ -32,11 +32,74 @@ import (
 	"k8s.io/api/core/v1"
 )
 
-// ExecuteApb - Runs an APB Action with a provided set of inputs
-func ExecuteApb(action string,
-	spec *Spec,
-	context *Context,
-	p *Parameters) (ExecutionContext, error) {
+// Executor - Main struct used for running APBs. Manages their lifecycle.
+type Executor struct {
```

Comment:
I would like to make it more clear that a single Executor for a single APB run. At least that is how I am reading this code am I wrong?

Reply:
Yeah that's exactly correct. What do you think the best way is to convey that?

Reply:
I would say documentation for the end user. Code examples will help too. I think that nothing from the code will make it more obvious. At least I can't think of anything at the moment.